pkg/operator/collection.go

```diff
@@ -354,7 +345,26 @@ func (r *collectionReconciler) ensureCollectorConfig(ctx context.Context, spec *
 	} else if err != nil {
 		return fmt.Errorf("update Prometheus config: %w", err)
 	}
-	return nil
+
+	// Reconcile any status updates.
+	var errs []error
+	for _, obj := range fullUpdates {
+		// Update does not patch status. Must patch status first because `Update` would mutate
+		// the `obj` parameter and reset it before we can patch it.
+		if err := patchMonitoringStatus(ctx, r.client, obj, obj.GetMonitoringStatus()); err != nil {
+			errs = append(errs, err)
+		}
+		if err := r.client.Update(ctx, obj); err != nil {
```

Comment:
Why do we need to update here? is it because we add `c.Spec.TargetLabels.Metadata`?

Reply:
Yea, "full" seems to mean "include non status updates" e.g. when defaulting.

I wonder if we can put better name here for those semantics, proposed in other comment.

Reply:
Check out the new `update` struct!